sources/logdog/src/log_request.rs

```diff
@@ -205,12 +218,114 @@ where
     Ok(())
 }
 
+/// Copies all file matching the glob pattern given by `request.instructions` to the tempdir with filename and path
+/// same as source file.
+fn handle_glob_request<P>(request: &LogRequest<'_>, tempdir: P) -> Result<()>
+where
+    P: AsRef<Path>,
+{
+    let glob_paths = glob(request.instructions).context(error::ParseGlobPattern {
+        pattern: request.instructions.to_string(),
+    })?;
+    for entry in glob_paths {
+        if let Ok(path) = entry {
+            if path.is_dir() {
+                // iterate the directory and copy files in the directory, nested
+                // sub-directories files will be copied only if matched by glob pattern.
```

Comment:
Grammar nit, but I also kind of don't understand what it means.

```suggestion
                // sub-directory files will be copied only if matched by glob pattern.
```